website/static/js/profile.js

```diff
@@ -122,6 +103,35 @@ var DateMixin = function() {
         self.endYear('');
         return true;
     };
+
+    self.startYear.extend({
```

Comment:
Yeah so since endYear and startYear rely on each other trying to have the `extend` calls in the same place as the initialization results in multiple TypeErrors.  Bumping down the `extend` calls until after both observables have been initialized lets you use both of these correctly without the errors.